docs/source/API/core/initialize_finalize/is_Initialized.rst

```diff
@@ -0,0 +1,56 @@
+``is_initialized``
+==================
+
+.. role::cpp(code)
+    :language: cpp
+
+Defined in header ``<Kokkos_Core.hpp>``
+
+Usage: 
+
+.. code-block:: cpp
+
+    Kokkos::is_initialized();
+
+
+Queries the initialization status of Kokkos and returns ``true`` if Kokkos is initialized and ``false`` is Kokkos is not initialized. This function can be called prior or after Kokkos initialization or finalization.
+
+Interface
+---------
+
+.. code-block:: cpp
+
+    Kokkos::is_initialized();
+    
+Requirements
+~~~~~~~~~~~~
+
+* ``Kokkos::is_initialized`` can be called before ``Kokkos::initialize``.
+* ``Kokkos::is_initialized`` can be called after ``Kokkos::finalize``.
+
+Semantics
+~~~~~~~~~
+
+* After calling ``Kokkos::initialize``, ``Kokkos::is_initialized()`` should return true.
+
+Example
+~~~~~~~
+
+.. code-block:: cpp
+
+    #include <Kokkos_Core.hpp>
+    #include <cassert>
+
+    int main(int argc, char* argv[]) {
+        assert(!Kokkos::is_initialized());
+        Kokkos::initialize(argc, argv);
+	assert(Kokkos::is_initialized());
+        Kokkos::finalize();
+        assert(!Kokkos::is_initialized());
```

Comment:
I am not sure if the last line here is correct ... does finalize really reset the initialization status? this would be problematic if multiple libs are possibly initializing kokkos as it could lead to re-initialization.

Reply:
It's likely wrong given we track two states: 
```
bool g_is_initialized      = false;
bool g_is_finalized        = false;
```